stl/inc/coroutine

```diff
@@ -40,7 +40,7 @@ struct _Coroutine_traits {};
 
 template <class _Ret>
 struct _Coroutine_traits<_Ret, void_t<typename _Ret::promise_type>> {
-    using promise_type = typename _Ret::promise_type;
+    using promise_type = _Ret::promise_type;

```

Comment:
Despite being added in C++20, we support this header in sub-c++20 modes with `/await:strict` to enable folks to migrate to the new/better/supported coroutine ABI independently of migrating their code to C++20. Let's revert this one:
```suggestion
    using promise_type = typename _Ret::promise_type;
```